### PR TITLE
Fix supervised pretraining bugs and add new datasets

### PR DESCRIPTION
**Pull request by theblackcat102**
**Code change**

Mainly typo and logic error by previous PR merge.

Code refactor in datasets

**Dataset change**

Added these datasets into training lists

- SODA
- Joke Explaination
- GSM8k

### CODE REVIEW

**Comment by theblackcat102** [comment]
And one big change was changing the question answer tokens to `<human>` and `<bot>` for better clarity. Since some of the datasets are about multi turns dialogue (SODA)

File: model/supervised_finetuning/configs/config.yaml [hunk 1]
```diff
@@ -18,7 +18,19 @@ defaults:
...
+    - trivia_qa_nocontext
+    - xsum
+    - cnn_dailymail
+    - prompt_dialogue
+    - multi_news
+    - scitldr
+    - soda
+    - joke
+    -
+    - joke
```
Comment (sanagno) [inline model/supervised_finetuning/configs/config.yaml:32]: typo?

**Review by sanagno — APPROVED** [review]
Thanks a lot for taking care of this! This was a pain.

File: model/supervised_finetuning/configs/config.yaml (hunk 1 above)
Comment (theblackcat102) [reply model/supervised_finetuning/configs/config.yaml]: Ah, yes it is, I accidentally added the config. Guess I will just update it to use my current datasets. These have been running for hours without issue on codegen 2B

**Comment by theblackcat102** [comment]
> Thanks a lot for taking care of this! This was a pain.

We should only allow PR with tests case written.

**Comment by theblackcat102** [comment]
Going to merge now

**Pull request merged**